core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -62,14 +62,18 @@ private[spark] class HeartbeatReceiver(sc: SparkContext)
 
   // "spark.network.timeout" uses "seconds", while `spark.storage.blockManagerSlaveTimeoutMs` uses
   // "milliseconds"
-  private val executorTimeoutMs = sc.conf.getOption("spark.network.timeout").map(_.toLong * 1000).
-    getOrElse(sc.conf.getLong("spark.storage.blockManagerSlaveTimeoutMs", 120000))
+  private val networkTimeoutS = Utils.timeStringAsS(sc.conf.get("spark.network.timeout", "120s"))
+  private val executorTimeoutMs = Utils.timeStringAsMs(
+    sc.conf.get("spark.storage.blockManagerSlaveTimeoutMs", 
+      s"${networkTimeoutS}s"))
```

Comment:
this doesn't preserve the order. Before `spark.network.timeout` takes higher precedence, and if that's not defined we default to `spark.storage.blockManagerSlaveTimeoutMs`, which is deprecated IIRC. Currently the priority is inverted.